### PR TITLE
Improve/Fix GUI version check - 5.0.1

### PR DESCRIPTION
**Pull request by retiutut**
- [x] Fix bug gui hangs on ping with no internet
- [x] Account for v5.0.0-alpha.0 and v5.0.0-beta.0 formats for GUI version check

### CODE REVIEW

File: OpenBCI_GUI/Extras.pde [hunk 1]
```diff
@@ -314,3 +314,24 @@ class TextBox {
...
+    int code = 200;
+    try {
+        URL siteURL = new URL(url);
+        HttpURLConnection connection = (HttpURLConnection) siteURL.openConnection();
+        connection.setRequestMethod("GET");
+        connection.setConnectTimeout(2000);
+        connection.connect();
+
+        code = connection.getResponseCode();
+        if (code == 200) {
```
Comment (retiutut) [inline OpenBCI_GUI/Extras.pde:328]: This is precisely what we want, to see the response code from an HTML webpage.

File: OpenBCI_GUI/Extras.pde [hunk 2]
```diff
@@ -391,3 +391,37 @@ class TextBox {
...
+        } else {
+            return false;
+        }
+    } catch (IOException e) {
+        return false;
+
+    }
+}
+
+public String getGUIVersionFromInternet(String _url) {
```
Comment (retiutut) [inline OpenBCI_GUI/Extras.pde:416]: Here is a simple method to fetch the JSON from Github Api and return version.

File: OpenBCI_GUI/TopNav.pde [hunk 3]
```diff
@@ -526,114 +526,81 @@ class TopNav {
     } //end mouseReleased
 
     //Load data from the latest release page from Github and the info.plist file
-    void loadGUIVersionData() {
+    void loadCompareGUIVersion() {
```
Comment (retiutut) [inline OpenBCI_GUI/TopNav.pde:529]: Cleaned up this method

File: OpenBCI_GUI/TopNav.pde [hunk 4]
```diff
@@ -526,114 +526,81 @@ class TopNav {
...
-
-        // Return the text in between
-        return s.substring(start, end);
+        return version;
     }
 
-    String removeV(String s) {
+    //Convert version string to float using each character as a digit.
+    //Examples: 5.0.0-alpha.2 -> 500.12, 5.0.1-beta.9 -> 501.29, 5.0.1 -> 501.5
+    private float getVersionAsFloat(String s) {
```
Comment (retiutut) [inline OpenBCI_GUI/TopNav.pde:576]: New consolidated method to parse strings and output a float as described. Should continue to work in perpetuity as long as existing version format is followed.

**Comment by daniellasry** [comment]
can you point out where the httprequests_processing library is used in the GUI code?

File: OpenBCI_GUI/TopNav.pde [hunk 5]
```diff
@@ -525,115 +525,82 @@ class TopNav {
...
-        // Move to the end of the beginning tag
-        // and find the index of the "after" String
-        start += before.length();
-        int end = s.indexOf(after, start);
-        if (end == -1) {
-            return "";
+    private String getGUIVersionFromInternet(String _url) {
+        String version = null;
+        try {
+            GetRequest get = new GetRequest(_url);
```
Comment (retiutut) [inline OpenBCI_GUI/TopNav.pde:564]: @daniellasry Here is the usage of the new HTTP request library. Much simpler than writing this method using Apache or Java methods, though this happens to use Apache in the background.

**Review by daniellasry — APPROVED** [review]

**Review by Andrey1994 — APPROVED** [review]

**Pull request merged**